boa/examples/jsarray.rs

```diff
@@ -0,0 +1,53 @@
+// This example goes into the details on how to pass closures as functions
```

Comment:
Looks good.
Would we move this into https://github.com/boa-dev/boa/pull/1161 once that's in?